test/unit/testDetectDataChange.js

```diff
@@ -246,6 +254,19 @@ describe('DetectDataChange.js', () => {
                 done();
             });
 
+            it('when executors whoDied tick boxes have not been changed', (done) => {
+                req.session.form.executors.list[1].isDead = true;
+                req.session.form.executors.list[1].isApplying = false;
+                req.session.form.executors.list[2].isDead = true;
+                req.session.form.executors.list[2].isApplying = false;
+                step.section = 'executors';
+                req.body = {executorsWhoDied: ['James Miller', 'Ed Brown']};
+                req.session.haveAllExecutorsDeclared = 'false';
+                const detectDataChange = new DetectDataChange();
+                expect(detectDataChange.hasDataChanged(ctx, req, step)).to.equal(false);
+                done();
+            });
+
```

Comment:
This test doesn't get as far as the new logic to test the failure as it needs multiple applicants :)

Reply:
👍 All done